### PR TITLE
[DM/FEATURE] Support thermal

### PR DESCRIPTION
**Pull request by GuEe-GUI**
[
Thermal drivers offer a generic mechanism for thermal management.
Usually it's made up of one or more thermal zones and cooling devices.
Each thermal zone contains its own temperature, trip points, and cooling devices.
All platforms with ACPI or OFW thermal support can use this driver.

We support the PWM cool fan drivers (GPIO will commit if test OK).

- BSP:
rockchip/rk3308
rockchip/rk3568

]

### 当前拉取/合并请求的状态 Intent for your PR

必须选择一项 Choose one (Mandatory):

- [ ] 本拉取/合并请求是一个草稿版本 This PR is for a code-review and is intended to get feedback
- [x] 本拉取/合并请求是一个成熟版本 This PR is mature, and ready to be integrated into the repo

### 代码质量 Code Quality：

我在这个拉取/合并请求中已经考虑了 As part of this pull request, I've considered the following:

- [x] 已经仔细查看过代码改动的对比 Already check the difference between PR and old code
- [x] 代码风格正确，包括缩进空格，命名及其他风格 Style guide is adhered to, including spacing, naming and other styles
- [x] 没有垃圾代码，代码尽量精简，不包含`#if 0`代码，不包含已经被注释了的代码 All redundant code is removed and cleaned up
- [x] 所有变更均有原因及合理的，并且不会影响到其他软件组件代码或BSP All modifications are justified and not affect other components or BSP
- [x] 对难懂代码均提供对应的注释 I've commented appropriately where code is tricky
- [x] 代码是高质量的 Code in this PR is of high quality
- [x] 已经使用[formatting](https://github.com/mysterywolf/formatting) 等源码格式化工具确保格式符合[RT-Thread代码规范](https://github.com/RT-Thread/rt-thread/blob/master/documentation/contribution_guide/coding_style_cn.md) This PR complies with [RT-Thread code specification](https://github.com/RT-Thread/rt-thread/blob/master/documentation/contribution_guide/coding_style_en.md)
- [x] 如果是新增bsp, 已经添加ci检查到[.github/workflows/bsp_buildings.yml](workflows/bsp_buildings.yml)  详细请参考链接[BSP自查](https://www.rt-thread.org/document/site/#/rt-thread-version/rt-thread-standard/development-guide/bsp-selfcheck/bsp_selfcheck)

### CODE REVIEW

File: components/drivers/include/dt-bindings/thermal/thermal.h [hunk 1]
```diff
@@ -0,0 +1,13 @@
+/*
+ * Copyright (c) 2006-2022, RT-Thread Development Team
+ *
```
Comment (Rbb666) [inline components/drivers/include/dt-bindings/thermal/thermal.h:3]: 加上日期，作者信息吧

File: components/drivers/thermal/SConscript [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+CPPPATH = [cwd + '/../include']
+
+src = ['thermal.c', 'thermal_dm.c']
+
+if GetDepend(['RT_THERMAL_COOL_PWM_FAN']):
+    src += ['thermal-cool-pwm-fan.c']
+
+group = DefineGroup('DeviceDrivers', src, depend = [''], CPPPATH = CPPPATH)
+
+for d in list:
```
Comment (Rbb666) [inline components/drivers/thermal/SConscript:20]: 这段代码是可以删掉的吧

Comment (GuEe-GUI) [reply components/drivers/thermal/SConscript]: 是的

File: components/drivers/include/dt-bindings/thermal/thermal.h (hunk 1 above)
Comment (GuEe-GUI) [reply components/drivers/include/dt-bindings/thermal/thermal.h:3]: 目前dts绑定头文件的默认都是不加这些信息的

**Comment by GuEe-GUI** [comment]
@Rbb666

**Review by Rbb666 — APPROVED** [review]

**Pull request merged**